pkg/platform/api/mono/mono.go

```diff
@@ -34,7 +37,32 @@ func Init(serviceURL *url.URL, auth *runtime.ClientAuthInfoWriter) *mono_client.
 	if auth != nil {
 		transportRuntime.DefaultAuthentication = *auth
 	}
-	return mono_client.New(transportRuntime, strfmt.Default)
+	client := mono_client.New(transportRuntime, strfmt.Default)
+
+	// For the Oauth client, prefer use of IPv4. This is needed particularly for device
+	// authorization, which compares the IP address of the State Tool request and the IP address of
+	// the web client request. If there's a mismatch, authorization fails. When this happens, it's
+	// often because the State Tool connects to the Platform via IPv6, but the browser does via IPv4
+	// (browsers apparently prefer IPv4 for now).
+
+	httpTransport, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		return client // for some reason the assertion can fail...
+	}
+
+	ipv4TransportRuntime := httptransport.New(serviceURL.Host, serviceURL.Path, []string{serviceURL.Scheme})
+	var dialer net.Dialer
+	ipv4Transport := httpTransport.Clone()
+	ipv4Transport.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+		if conn, err := dialer.DialContext(ctx, "tcp4", addr); conn != nil {
```

Comment:
```suggestion	
	ipv4Transport := httpTransport.Clone()
	ipv4Transport.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
		dialer := &net.Dialer{}
		if conn, err := dialer.DialContext(ctx, "tcp4", addr); conn != nil {
```

Reply:
What you have might work but it's not a pattern I've seen before and I'm inclined to stick with what I know. From a quick reference this also seems to be how net.Dialer is normally used. Happy to be educated though.

Reply:
I thought it was expensive to create a `net.Dialer` each time a DialContext is used, but perhaps this doesn't happen very often.

Reply:
You can't call `DialContext` without creating it anyway. I think your `var declaration previously on line 51 was effectively doing this for you.